test/server/configuration_impl_test.cc

```diff
@@ -327,7 +357,10 @@ TEST(ConfigurationImplTest, NullTracerSetWhenTracingConfigurationAbsent) {
 
   NiceMock<Server::MockInstance> server;
   server.local_info_.cluster_name_ = "";
-  MainImpl config(server);
+  Upstream::ProdClusterManagerFactory cluster_manager_factory(
+      server.runtime(), server.stats(), server.threadLocal(), server.random(), server.dnsResolver(),
+      server.sslContextManager(), server.dispatcher(), server.localInfo());
+  MainImpl config(server, cluster_manager_factory);
```

Comment:
Not sure I follow -- by multiple constructors do you mean offer `MainImpl(Server::Instace&)` as well as the new `MainImpl(Server::Instance&, ClusterManagerFactory&)`? That might work but I'm not sure about the lifetimes -- the `MainImpl` only holds a reference to the `ClusterManagerFactory` (and needs it in `initialize()`, not in its ctor) so I don't immediately see a clean way but maybe I'm missing it.

If the concern is just that these tests are noisy now, I agree -- if I refactored them to use a test fixture, so that we only had to say that once and the individual tests were more compact, would that suit you?